docs/user-guide/spark-performance.md

```diff
@@ -0,0 +1,74 @@
+
+# Spark Application Performance
+
+This document describes the test framework used internally to test the CQL on Spark application, the performance we observed during testing, and introduces some techniques for monitoring and tuning parameters that will affect runtime. Included at the end are some tips for CQL authors on how to create CQL queries that perform better than others.
+
+## Test Setup
+
+Apache Spark 3.1.2 / Hadoop 3.2 running in IBM Cloud under IBM Kubernetes (K8S) Service
+1 K8S cluster, 18 worker nodes
+	4 core, Intel Xeon Processor (Skylake, IBRS) 2.6 ghz
+	16 GB RAM
+
+	spark.default.parallelism           50
+	spark.driver.memory                 1G
+	spark.driver.memoryOverhead         1G
+	spark.executor.memory               4G
+	spark.executor.memoryOverhead       1G
+	spark.eventLog.enabled              true
+	spark.executor.instances            4
+
+Data stored in deltalake format under IBM Cloud Object Store (COS)
+
+	Region-local COS bucket
+	Direct endpoints used between K8S cluster and COS
+
+2 aggregation contexts
+
+	context 1:
+		245,000 records joined 1:n with three related tables
+			join1    32,000,000
+			join2    25,000,000
+			join3    80,000,000
+	context 2: 
+		24,000,000 records, no joins
+
+5 CQL expressions in 5 libraries
+	- No use of valuesets
+	
+	2 CQL expressions for context 1
+	3 CQL expressions for context 2
+
+## Performance
+
+![performance](HI_Pilot_Performance_SmallTenantHistory.png)
+
+## IO
+
+Each data table is read from disk each time it is included in a context definition. If the same table is used in more than one context definition, it will be read more than one time. Most of the CQL logic we've been testing with has been fairly simple and the majority of the program's runtime is spent reading and regrouping data. If there is a need to reduce the overall runtime of a single program execution, users might find it beneficial to reduce the number of context definitions included in a single run. This can be one using the `-a` program argument for a single context definition file or by simply dividing the context definitions into separate files and referencing the appropriate file in the `-d` program option for each execution.
```

Comment:
This can be 'd'one using the `a` program argument for a single context definition file .....  (last statement in the paragraph) missing d for done.